api/v1beta1/metallb_webhook.go

```diff
@@ -0,0 +1,65 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func (metallb *MetalLB) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(metallb).
+		Complete()
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-metallb-io-v1beta1-metallb,mutating=false,failurePolicy=fail,groups=metallb.io,resources=metallbs,versions=v1beta1,name=metallbvalidationwebhook.metallb.io,sideEffects=None,admissionReviewVersions=v1
+
+var _ webhook.Validator = &MetalLB{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for MetalLB.
+func (metallb *MetalLB) ValidateCreate() error {
+	return metallb.validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for MetalLB.
+func (metallb *MetalLB) ValidateUpdate(old runtime.Object) error {
+	return metallb.validate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for MetalLB.
+func (metallb *MetalLB) ValidateDelete() error {
+	return nil
+}
+
+func (metallb *MetalLB) validate() error {
+	for _, ct := range metallb.Spec.ControllerTolerations {
+		if ct.TolerationSeconds != nil && *ct.TolerationSeconds > 0 && ct.Effect != v1.TaintEffectNoExecute {
+			return errors.New("ControllerToleration effect must be NoExecute when tolerationSeconds is set")
+		}
+	}
+	for _, ct := range metallb.Spec.SpeakerTolerations {
```

Comment:
Can you add a validation for the weight field of node affinities (and other, if any) to be between 1 and 100 ?

Reply:
done